src/main/sensors/battery.c

```diff
@@ -221,41 +220,42 @@ static void updateCurrentDrawn(int32_t lastUpdateAt)
 
 void updateCurrentMeter(int32_t lastUpdateAt, rxConfig_t *rxConfig, uint16_t deadband3d_throttle)
 {
-    switch(batteryConfig->currentMeterType) {
-        case CURRENT_SENSOR_ADC:
-            updateBatteryCurrent();
-
-            updateCurrentDrawn(lastUpdateAt);
-
-            break;
-        case CURRENT_SENSOR_VIRTUAL:
-            amperage = (int32_t)batteryConfig->currentMeterOffset;
-            if (ARMING_FLAG(ARMED)) {
-                throttleStatus_e throttleStatus = calculateThrottleStatus(rxConfig, deadband3d_throttle);
-                int throttleOffset = (int32_t)rcCommand[THROTTLE] - 1000;
-                if (throttleStatus == THROTTLE_LOW && feature(FEATURE_MOTOR_STOP)) {
-                    throttleOffset = 0;
+    #ifdef USE_ESC_SENSOR
+    if (feature(FEATURE_ESC_SENSOR))
```

Comment:
I don't think that ignoring `batteryConfig->currentMeterType` when `FEATURE_ESC_SENSOR` is enabled is the right thing to do. There are plenty of reasons why users would want to use the ESC sensor only for vBat or only for current. Also, maybe one day ESC feedback will even support things like RPM or temperature, and this will become even more true.